### DISCUSSION TITLE
BT815 with NHD-7.0 and ESP32

### DISCUSSION THREAD

**Discussion by Qtakruri** [RudolphRiedel/FT800-FT813#87] [Q&A]
Hi, 

I am trying to run [NHD-7.0-800480EF-ASXV#-CTP] with BT815 through an ESP32-Wroom32E. I uploaded the code to [NHD-7.0-800480FT-CSXV-CTP] board that has FT813 and it works fine. but for some reason, it isn't working with the BT815 chip, I have my schematics built according to the BT815 datasheet and double-checked the components. 
it may be helpful to know that I am seeing all the signals out from the BT815 chip like (HSYNC,VSYNC,PCLK,etc..) 
are there any changes I need to make in the code to have the BT815 chip run the screen?

**Comment by RudolphRiedel** [comment]
Hello,

so you have a custom PCB that is designed following perhaps the schematic NHD-7.0-800480FT-CSXV-CTP and the datasheet of the BT815?
The panel NHD-7.0-800480EF-ASXV#-CTP is the one that is used for the NHD-7.0-800480FT-CSXV-CTP.

The BT815 board should work with the EVE_NHD_70 configuration.
Not with BT815 features though and "only" with 60MHz, but for a start it should work.

And a new configuration to really make use of the BT815 would need this changed:
#define EVE_GEN 3
Or simply use EVE_RiTFT70.

I am using an EVE_RVT50H right now which has a BT817.
I modified the profile in EVE_config.h to "#define EVE_GEN 2" and it did not compile anymore.
I commented out the one BT81x specific line from my display function and it works.

So using a BT81x like it is a FT81x works fine, they "only" added more features.
At least when the board already works with the BT81x.

This looks like a hardware issue.
Is this an extra panel? Have you tried using it with the FT813 board?
And double and tripple check your FFC connectors for solder bridges.

**Comment by Qtakruri** [comment]
Hey Rudolph,

Thank you for the reply. I just wanted to cross software issues from my checklist, I’m going to take a deeper look into the hardware side.

I am using a breakout board<https://www.adafruit.com/product/4905> for the FFC connector and running wires from my PCB to the FFC board, but I don’t think it’s the problem. What’s really confusing is that I see clean signals coming from the BT815 chip, but the screen is blank the BL is not even turning on, even though I can see 25.4V across the LEDA and LEDK pins

**Comment by RudolphRiedel** [comment]
>I am using a breakout board<https://www.adafruit.com/product/4905> for the FFC connector and running wires
>from my PCB to the FFC board, but I don’t think it’s the problem.

Oh, at 30MHz it might be the issue, do you have series resistors?

> What’s really confusing is that I see clean signals coming from the BT815 chip, but the screen is blank the BL is not even turning on,

Have you tried shining a light on the screen to check if anything is displayed?

>even though I can see 25.4V across the LEDA and LEDK pins

That is more than twice what the backlight is documented to need, according to the datasheet the maximum is 10.2V.

**Comment by Qtakruri** [comment]
I am using a breakout boardhttps://www.adafruit.com/product/4905<https://linkprotect.cudasvc.com/url?a=https%3a%2f%2fwww.adafruit.com%2fproduct%2f4905&c=E,1,VC49hsn_o0kFlCipTanUaWhh2XWxmad56NXEurbaJoGTaJZv_P6GHEv_CnTnDEkDzyA67clX4GLtefkcsQCIn1hcTvw9TSk5UGaEoVayggzvVHHClv7a&typo=1> for the FFC connector and running wires
from my PCB to the FFC board, but I don’t think it’s the problem.

Oh, at 30MHz it might be the issue, do you have series resistors?

Yes I have 33OHM resistors inline with the RGB signals. I removed the wires and plugged the breakout board directly to the pins to reduce any signal loses.

What’s really confusing is that I see clean signals coming from the BT815 chip, but the screen is blank the BL is not even turning on,

Have you tried shining a light on the screen to check if anything is displayed?

Yes and I cant see anything

even though I can see 25.4V across the LEDA and LEDK pins

That is more than twice what the backlight is documented to need, according to the datasheet the maximum is 10.2V.

I removed the LED driver circuit and applied 10V on the LEDK and LEDA pins directly from a power source, still nothing happened.



—
Reply to this email directly, view it on GitHub<https://github.com/RudolphRiedel/FT800-FT813/discussions/87#discussioncomment-5982431>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AZRQAWN34K3O75U4RPLQF3LXHUN2ZANCNFSM6AAAAAAYMAFGUI>.
You are receiving this because you authored the thread.Message ID: ***@***.******@***.***>>
CAUTION: This email originated from outside of the organization. Do not click links or open attachments unless you recognize the sender and know the content is safe.


The information in this e-mail is confidential and may be legally privileged; it is intended solely for the addressee. If you have received this communication in error, you are notified that disclosing, copying, distributing, or taking any action in reliance on the contents of this information is strictly prohibited and please notify the sender immediately. Although the company has taken reasonable precautions to ensure no viruses are present in this email, the company cannot accept responsibility for any loss or damage arising from the use of this email or attachments.

**Comment by RudolphRiedel** [comment]
I was about to ask if there is a chance that the polarity on the FFC 40 is wrong but I looked at a foto and it even is printed on the cable.
What I do wonder though is what that PCB on the FFC is doing. It looks like there is a swiching regulator on there, that may be the step-up for the backlight.
Have you measured the voltage across pins 1 and 2 of the NHD-7.0-800480FT-CSXV-CTP?

I have no further ideas to resolve this given the limited amount of data and my very limited experience with the signals on the panel side of the chips.

**Comment by Qtakruri** [comment]
Hey Rudolph,

Sorry it took a while to get back to you. I was trying different solutions with the hardware but without any luck.

I have a backlight driver on my board (Copied from BT815 datasheet) and I measured the voltage on pins 1 and 2 of the ribbon connector and it was 25.4, but after you pointed out that the voltage for the screen is at max 10V I removed the step up circuit and applied 9.5V on pin 1 and 2 and still nothing.

I attached a PDF of my schematic for the screen. Maybe you’ll see something I am not seeing.

Also, I want to thank you for helping me on this, I appreciate every minute of your time.

Thanks,

**Comment by RudolphRiedel** [comment]
My question was what the voltage is at the backlight pin of the NHD-7.0-800480FT-CSXV-CTP, according to the datasheet you should measure around 10V.
But it is possible that there are only 5V or 3.3V if the backlight driver is already in the panel and not on the FT813 board.
I only have a NHD-3.5-320240FT-CSVX-CTP and after buying this one I did not want to get any other from NewHaven.
And then they never went past FT813.
